File: frontend/src/lib/schemas/userSchema.ts
import { z } from "zod";

export const userDTO = z.object({
  user_id: z.string().uuid(),
  username: z.string().min(3),
  email: z.string().email(),
});

export const userUpdateDTO = z.object({
  username: z.string().optional(),
  email: z.string().email().optional(),
  oldPassword : z.string().optional(),
  newPassword: z.string().optional(),
});

export type UserUpdateInput = z.infer<typeof userUpdateDTO>;
export type UserInput = z.infer<typeof userDTO>;

File: backend/src/modules/user/user_schema.py
# External Exports
from pydantic import BaseModel, EmailStr
from typing import Optional
from uuid import UUID
from datetime import datetime


class UserSchema(BaseModel):
    id: UUID
    username: str
    email: EmailStr

    class Config:
        from_attributes = True

class UserCreateSchema(BaseModel):
    username: str
    email: EmailStr
    password: str

class UserUpdateSchema(BaseModel):
    username: Optional[str] = None
    email: Optional[EmailStr] = None
    oldPassword: Optional[str] = None 
    newPassword: Optional[str] = None